Assets/src/Components/Board/BoardMotion.cs

```diff
@@ -0,0 +1,342 @@
+﻿using Assets.src.Enums;
+using System.Linq;
+using UnityEngine;
+using Assets.src.Models;
+using System.Collections.Generic;
+using Assets.src.Helpers;
+
+public class BoardMotion : BoardBase
+{
+    public GameObject SnakeStep;
+    public GameObject Target;
+
+	// Use this for initialization
+	void Start ()
+    {
+		
+	}
+	
+	// Update is called once per frame
+	void Update ()
+    {
+        var direction = InputToDirection();
+        
+        if (direction != DirectionEnum.None)
+        {
+            var stepData = GetLastSnakeStepData();
+            var position = stepData.GetComponent<SnakeStepPosition>();
+
+            var isWallClosed = IsWallOnWay(position.ColumnPosition, position.RowPosition, direction);
+            if (isWallClosed)
+            {
+                return;
+            }
+
+            if (stepData.Number == 1)
+            {
+                RemoveAllSteps();
+                RemoveTarget();
+
+                AddNewStep(
+                    position.ColumnPosition,
+                    position.RowPosition,
+                    stepData.StartNumber + 2,
+                    stepData.StartNumber + 1,
+                    direction
+                );
+                GenerateTarget(Target);
```

Comment:
It is not good to Instantiate objects in Update method. Better to use pool of objects, and store objects on scene in off state